sys/include/net/sock/dtls.h

```diff
@@ -452,13 +474,14 @@ void sock_dtls_init(void);
  *                      @ref sock_udp_create()to be used underneath.
  * @param[in] tag       Credential tag of the sock. The sock will only use
  *                      credentials with the same tag given here.
- * @param[in] method    Defines the method for the client or server to use.
+ * @param[in] method    Defines the role of the endpoint and the DTLS version
+ *                      to use.
  *
  * @return  0 on success.
  * @return  -1 on error
  */
 int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
-                     credman_tag_t tag, unsigned method);
+                     credman_tag_t tag, sock_dtls_method_t method);
```

Comment:
This may be premature optimization, but could you (with your `tinydtls` implementation) check how redefining this to

```C
int sock_dtls_create(sock_dtls_t *sock, sock_udp_t *udp_sock,
                     credman_tag_t tag, unsigned version, unsigned role);
```

works out memory-wise? Compared to this? I think from the user experience this would be better, but I'm not sure if this might be more memory-friendly (if it only safes a few bytes or so (<10B), I would still prefer dividing up the values).

Reply:
With `sock_dtls_method_t`, the compiled size of the example application is 20 bytes more compared to the separate version and role. So it is clearly in favor of the separate version. I'll change this.